### DISCUSSION TITLE
Tycho build - org.eclipse.m2e.maven.runtime Unresolved requirement javax.annotation

### DISCUSSION THREAD

**Discussion by to-fr** [eclipse-m2e/m2e-core#1135] [Q&A] [answered]
Hi,

with the latest Eclipse 2022-12 containing
  M2E - Maven Integration for Eclipse	2.1.2.20221130-2239

I get the following error:

```Unresolved requirement: Require-Bundle: org.eclipse.m2e.core
    -> Bundle-SymbolicName: org.eclipse.m2e.core; bundle-version="2.0.5.20221128-1757"; singleton:="true"
       org.eclipse.m2e.core [11]
         Unresolved requirement: Require-Bundle: org.eclipse.m2e.maven.runtime; bundle-version="[3.8.6,4.0.0)"
           -> Bundle-SymbolicName: org.eclipse.m2e.maven.runtime; bundle-version="3.8.604.20221130-2149"
              org.eclipse.m2e.maven.runtime [12]
                Unresolved requirement: Import-Package: javax.annotation; version="[1.2.0,2.0.0)"
```

I am using Tycho 2.7.5. With the previous Eclipse 2022-09 the error does not occur.

I already added
`<unit id="javax.annotation" version="0.0.0"/>`
to the target definition which is resolved to
`p2.eclipse.plugin:javax.annotation:jar:1.3.5.v20221112-0806:system`

I also tried using the latest snapshot versions for m2e but the error remains.

What could be the problem here?

**Comment by mickaelistria** [comment]
Can you please provide minimal steps to rerproduce from a fresh workspace?

**Comment by laeubi** [comment]
I have looked into this and I think this is not a problem with m2e directly (as the same build fails even without m2e) but that the update-site / features of m2 do not contain everything we depends on (in this case `javax.annotations`).

**Comment by laeubi** [comment, marked as answer]
I checked that the build succeeds with `mvn clean install -Dtycho-version=3.0.1` so this issue seems to be cause with **Tycho 2.7.5** no longer able to resolve newer platform releases!

I therefore suggest to upgrade to Tycho 3.0.1 in the build.
